test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ObjectPath.java

```diff
@@ -148,4 +149,14 @@ private Object evaluate(String key, Object object, Stash stash) throws IOExcepti
 
         return list.toArray(new String[list.size()]);
     }
+
+    public XContentBuilder toXContentBuilder(XContent xContent) throws IOException {
```

Comment:
is this used in one single place? it seems specialized to what we need in insertIntoXContent, maybe it should be part of that method?